phoenix-core-client/pom.xml

```diff
@@ -438,5 +438,9 @@
       <groupId>org.hdrhistogram</groupId>
       <artifactId>HdrHistogram</artifactId>
     </dependency>
+    <dependency>
```

Comment:
Is it OK to add this dependency ? @stoty can you confirm ?

Reply:
Having phoenix-core-client depend on hbase-server is really bad practice because it ultimately kills the whole purpose of having two separate phoenix-core modules, one for client and one of server.

But the bigger problem here is that we use the same iterator class at both client and server side, and depending on which side uses it, the context like this can be used.

It would have been much simpler if we had iterators of type `ResultIterator` only being used by clients and scanners of type `BaseRegionScanner` only being used by server. Then we could make `ScannerContext` available to only scanners of type `BaseRegionScanner`.

Reply:
Now i think probably we might still be able to get around this by not passing scanner context to `OrderedResultIterator` and `OffsetResultIterator`?

Reply:
Can you point me where this is a problem in the code ?
There are too many change files for me to check easily.

Reply:
@stoty initial implementation had introduced a dependency on hbase-server in phoenix-client since I was passing ScannerContext through `OrderedResultIterator` and `OffsetResultIterator`, which was not acceptable. I have now tried an alternate implementation where I avoid doing this.